contracts/bonding/BondingManager.sol

```diff
@@ -839,18 +873,37 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
             earningsPool.setStake(t.earningsPoolPerRound[lastUpdateRound].totalStake);
         }
 
+        if (treasuryBalanceCeiling > 0) {
+            uint256 treasuryBalance = livepeerToken().balanceOf(treasury());
+            if (treasuryBalance >= treasuryBalanceCeiling) {
+                // halt treasury contributions until the cut rate param is updated again
+                treasuryRewardCutRate = 0;
```

Comment:
@yondonfu this may be a philosophical discussion better suited for the forum, but I think the main reason we set it to zero was in the spirit of the "ice age" - where it forced a community to come together to take governance action at an important moment needing review. Rather than the ceiling being some important balance limit that needs to be maintained. IE if the ceiling gets hit in the first place, it means the treasury isn't successfully allocating funds, and should be reviewed, or at least funding put on pause until it can justify that it needs more.

Reply:
Got it. Thanks for the clarification. With that in mind, I think the current logic is fine.